soroban-env-macros/src/synth_dispatch_host_fn_tests.rs

```diff
@@ -220,11 +220,20 @@ pub fn generate_hostfn_call_with_wrong_types(file_lit: LitStr) -> Result<TokenSt
     let mut type_to_fn_arg: BTreeMap<String, Vec<(Function, usize)>> = BTreeMap::new();
     for m in root.modules.clone() {
         for f in m.functions.clone() {
-            for (i, a) in f.args.iter().enumerate() {
-                type_to_fn_arg
-                    .entry(a.r#type.clone())
-                    .or_default()
-                    .push((f.clone(), i));
+            // checks if the current ledger protocol version falls between
+            // supported protocol versions of this function
+            if !(f
+                .min_supported_protocol
+                .is_some_and(|v| v > LEDGER_PROTOCOL_VERSION)
+                || f.max_supported_protocol
```

Comment:
This is pretty hard to read: `||` looks like a closure at a first glance and there is also a negation of the whole complex expression. If you go for the helper, as suggested above, I would suggest breaking this down in a couple lines.

Reply:
fixed